### PR TITLE
Upgrade bedrock mongodb 7

### PR DESCRIPTION
**Pull request by aljones15**
NOTE: this branch is being used in a project currently.
Pushing to this Pr could create instability.

This should be better than: https://github.com/digitalbazaar/bedrock-ledger-agent/pull/30
when finished and also it contains a fix compatible with the removal of bedrock-identity.

- [x] review the `bedrock-identity` removal code
- [x] review the mongodb upgrades
- [x] Update the test project (related PR: https://github.com/digitalbazaar/bedrock-ledger-agent/pull/28)
- [x] use this branch in a working project with all related libs in the release.
- [x] MERGE IN PR WITH FIXED TESTS https://github.com/digitalbazaar/bedrock-ledger-agent/pull/35
- [ ] Swap out branches for releases on approval

Related PRs:
- [bedrock-ledger-node PR](https://github.com/digitalbazaar/bedrock-ledger-node/pull/73)
- [bedrock-ledger-consensus-continuity-storage PR](https://github.com/digitalbazaar/bedrock-ledger-consensus-continuity-storage/pull/6)
- [bedrock-ledger-storage-mongodb PR](https://github.com/digitalbazaar/bedrock-ledger-storage-mongodb/pull/51)
- [bedrock-ledger-agent PR](https://github.com/digitalbazaar/bedrock-ledger-agent/pull/34)
- [bedrock-ledger-core PR](https://github.com/digitalbazaar/bedrock-ledger-core/pull/9)
- [bedrock-ledger-consensus-continuity PR](https://github.com/digitalbazaar/bedrock-ledger-consensus-continuity/pull/202)
- [bedrock-ledger-consensus-uni PR](https://github.com/digitalbazaar/bedrock-ledger-consensus-uni/pull/8)

### CODE REVIEW

**Comment by gannan08** [comment]
Needs merge.

cc: @dlongley @mattcollier

**Review by gannan08 — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -28,18 +29,18 @@
...
-    "bedrock-permission": "^2.3.2",
+    "bedrock-mongodb": "7.x - 8.x",
+    "bedrock-passport": "3.x - 6.x",
+    "bedrock-permission": "^3.0.0",
     "bedrock-rest": "2.1.0 - 3.x",
     "bedrock-server": "^2.2.0"
   },
   "engines": {
-    "node": ">=8"
+    "node": ">=10"
```
Comment (aljones15) [inline package.json:43]: @mattcollier I believe this needs to be node 12 now.

Comment (mattcollier) [reply package.json:43]: I think at this point in time we would just change it to 14.

Comment (aljones15) [reply package.json:43]: Ok I will do that after your pending PR is merged into this one. I dogwood aware of your code coming into this branch?

**Review by mattcollier — APPROVED** [review]
I'm comfortable with merging this, but not making a release.  For one thing, there are still many dev deps in package files which are already being addressed in follow-on PRs.

Thanks @aljones15 for putting this together.

**Review by dlongley — APPROVED** [review]
Rubberstamping, thanks!

**Pull request merged**